### DISCUSSION TITLE
JSON post body parsed as string

### DISCUSSION THREAD

**Discussion by vjau** [fastify/fastify#5377] [Q&A] [answered]
Hi everyone,
i have an existing fastify app with a post route that i call with a regular SPA. It works correctly.
I'm trying to develop a chrome extension. The calling code using fetch has been copied and pasted from my working spa. It is posting to the same fastify server i'm using for my SPA.
The only difference is that the fetch on the SPA was using CORS, and in the chrome extension, i set it up to no-cors, since CORS is not needed in a chrome extension.
Now, the problem i see, is that post validation, the JSON body is parsed as a string instead of an object. It is as if the JSON whas not parsed at all. This cause the validation to fail. 
I have tried to peek in the request sent by the extension in a "onrequest" hook, but since the body has not been parsed at this stage, it is undefined.
Where could i go from here ? 

Thank you.

**Comment by Eomm** [comment, marked as answer]
I would check the headers - it seems the `content-type` is missing + there is the `preValidation` hook

**Reply by vjau** [reply]
Thank you for your help. 
You are right, the `content-type` is wrong, it became  `text/plain;charset=UTF-8`.
Somehow, the chrome extension is messing with it since i set it up correctly in my fetch request.
I will investigate further on the chrome front. 
Thank you !

**Reply by vjau** [reply]
For those interested, i found out you can't do a `no-cors` request with `application/json` with fetch ! 😱
https://github.com/JakeChampion/fetch/issues/318#issuecomment-249573195
